tools/azsdk-cli/specs/1-env-setup-env-verification.spec.md

```diff
@@ -0,0 +1,576 @@
+- [Testing Strategy](#testing-strategy)
+
+---
+
+## Definitions
+
+**Environment Setup**: The process of ensuring all required tools, SDKs, and dependencies are installed and properly configured for SDK development work.
+
+**Requirements**: Language-specific and core tools needed to successfully run Dev Inner Loop MCP tools (e.g., TypeSpec compiler, language SDKs, build tools).
+
+**Verification Check**: A command or script that determines whether a specific requirement is installed and accessible in the developer's environment.
+
+**Lazy Loading**: Checking only the requirements for the language(s) currently being worked on, rather than checking all languages upfront.
+
+**Source of Truth**: A single, authoritative configuration that defines what requirements are needed for each language and how to install them.
+
+---
+
+## Background / Problem Statement
+
+### Current State
+
+**Current state per language:**
+
+- **.NET**: Developers manually discover missing dependencies when build or generation commands fail. No proactive verification.
+- **Java**: Same as .NET - reactive discovery through failures.
+- **JavaScript**: Same as .NET - reactive discovery through failures.
+- **Python**: Some repos have `azpysdk` which can be checked, but no unified verification across all required tools.
+- **Go**: Same as .NET - reactive discovery through failures.
+
+**Data plane vs. Management plane:**
+
+Both data plane and management plane SDK development currently lack proactive environment verification. TypeSpec-based generation (common in data plane) and Swagger-based generation (common in management plane) each have their own tooling requirements, but neither has automated verification.
+
+### Why This Matters
+
+Without proactive environment verification:
+
+1. **Poor Developer Experience**: Developers waste time debugging cryptic error messages that are actually due to missing tools
+2. **Friction for New Contributors**: New team members or open-source contributors struggle to get started
+3. **MCP Agent Limitations**: When the agent attempts to run tools and they fail due to missing dependencies, the agent can't distinguish between actual errors and setup issues
+4. **Context Switching Pain**: When developers work across multiple language repos, they must manually remember and install different requirements for each language
+
+**Agent vs. Manual Workflow:**
+
+- **With Agent (Current)**: Agent attempts a command → fails → developer must interpret error → manually install missing tool → retry
+- **With Agent (Proposed)**: Agent checks setup first → proactively tells developer what's missing and how to fix it → developer installs → proceed with confidence
+- **Manual CLI (Current)**: Developer runs command → fails → searches documentation or asks teammates → installs tool → retry
+- **Manual CLI (Proposed)**: Developer explicitly runs `azsdk verify-setup` → sees clear list of missing items → follows install instructions → proceeds
```

Comment:
Echoing my main comment, I think `--verify-setup` as a flag for a command will be easier to reason about for users. If I'm running some command then I can check the setup verification for that command or for the parent command group (so `azsdk package --verify-setup` could validate for the language a package is being generated for, etc.).